manim/mobject/text/tex_mobject.py

```diff
@@ -399,34 +424,39 @@ def set_opacity_by_tex(
         """
         if remaining_opacity is not None:
             self.set_opacity(opacity=remaining_opacity)
-        for part in self.get_parts_by_tex(tex):
+        for part in self.get_parts_by_tex(tex_string):
             part.set_opacity(opacity)
         return self
 
-    def set_color_by_tex_to_color_map(self, texs_to_color_map, **kwargs):
-        for texs, color in list(texs_to_color_map.items()):
+    def set_color_by_tex_to_color_map(
+        self,
+        texes_to_color_map: dict[str | Iterable[str], ParsableManimColor],
+        **kwargs: Any,
+    ) -> Self:
+        for texes, color in list(texes_to_color_map.items()):
             try:
                 # If the given key behaves like tex_strings
-                texs + ""
-                self.set_color_by_tex(texs, color, **kwargs)
+                texes + ""  # type: ignore [operator]
+                self.set_color_by_tex(texes, color, **kwargs)  # type: ignore [arg-type]
             except TypeError:
                 # If the given key is a tuple
-                for tex in texs:
+                for tex in texes:
                     self.set_color_by_tex(tex, color, **kwargs)
```

Comment:
Probably the same feedback as before, this feels pretty hacky to me.

Reply:
Same as before: I decided to implement your proposal.